### ISSUE TITLE
Add contributing instructions?

### ISSUE THREAD

**Issue by neezer** [streamich/memfs#7]
I've started using this library for writing tests for my `fs` operations and I'm starting to notice some shortcomings of this library. Namely,

- [Nodes can never be detected as symlinks using `lstat`](https://github.com/streamich/memfs/blob/master/index.ts#L66-L68). This makes testing `fs` logic dealing with symlinks, well, impossible.
- [If a node exists, the error thrown](https://github.com/streamich/memfs/blob/master/index.ts#L221) does not conform to standard [Node System Errors](https://nodejs.org/api/errors.html#errors_system_errors), making it hard to write logic that works with both the native `fs` module and `memfs`.

I initially opened up a [PR](https://github.com/streamich/memfs/pull/6) to fix one, but realized that this project was written in TypeScript, which I haven't worked with yet.

I tried installing TypeScript and compiling and was greeted with errors about missing type definitions and whatnot. I've already run `npm install`, but I think the type definitions are out-of-date or not configured correctly. As the last activity on this repo was 8 months ago, I think there might have been some drift in the interim.

Anywho, I'd love to make some contributions to this repo, but I can't afford to take the time trying to reverse-engineer the dev setup. Mind adding some dev instructions to the README so I can skip to making PRs?

**Comment by streamich** [comment]
There is no specific dev setup for TypeScript. The simplest to get started is to download the [Webstorm](https://www.jetbrains.com/webstorm/) IDE and when you open a TypeScript file for the first time, it offers you to start TypeScript compiler that will compile to .js as you type.

On a side note, I've been writing `fs` module clone, you can copy some error messages from [here](https://github.com/streamich/full-js/blob/master/src/lib/fs.ts#L26-L55).

**Comment by megamindbrian** [comment]
This is what I used in combination with mock-require to get angular-cli to work with typescript:
Unfortunately typescript doesn't use the standard FS module from Node.

```
var ts = require('typescript');

var tsMock = Object.assign({}, ts);
var toMock = {
    readFile: 'readFileSync',
    realpath: 'realpathSync',
    writeFile: 'writeFileSync',
    fileExists: 'existsSync',
    directoryExists: 'existsSync',
    createDirectory: 'mkdirSync',
    getDirectories: 'readdirSync'
};
Object.keys(toMock)
    .forEach(k => {
        if(typeof fsMock[toMock[k]] === 'function') {
            tsMock.sys[k] = function () {
                // typescript.sys functions return defined on error
                try {
                    return fsMock[toMock[k]].apply(memfs, arguments);
                } catch (e) {
                    return;
                }
            };
        }
    })

mock('typescript', tsMock);
```

**Comment by megamindbrian** [comment]
I think I need symlinks to be able to use NPM with this.

**Comment by streamich** [comment]
I've released a new version 2.0, which now has symlink support. Also the errors thrown should now be the same (or almost) as the ones Node throws, keeping in mind that Node itself is now in transition of moving to its new error constructors in `internals/errors.js`.

Contributing now should be simple as well:

    git clone https://github.com/streamich/memfs
    npm install

Edit `./src` folder. Install `ts-node` globally, if you want to run, for example:

    ts-node demo/toJSON.ts

Then

    npm test
    npm run build

And submit a PR.
